### PR TITLE
[e2e] test full python version in installer tests

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Assert we install the expected embed python version at install

### Motivation

We recently had a bug on macos where the build pipeline was not embedding the expected python version. This improves tests by adding the full python version test at install time.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by clarkb7 — APPROVED** [review]
thank you!

**Review by richl-dd — APPROVED** [review]

File: test/new-e2e/tests/agent-platform/common/agent_behaviour.go [hunk 1]
```diff
@@ -192,23 +192,34 @@ func CheckDogstatsdAgentRestarts(t *testing.T, client *TestClient) {
...
-// CheckAgentPython runs tests to check the agent use the correct python version
-func CheckAgentPython(t *testing.T, client *TestClient, version string) {
-	t.Run(fmt.Sprintf("set python version %s and restarts", version), func(tt *testing.T) {
+const (
+	// ExpectedPythonVersion2 is the expected python 2 version
+	// Bump this version when the version in omnibus/config/software/python2.rb changes
+	ExpectedPythonVersion2 = "2.7.18"
+	// ExpectedPythonVersion3 is the expected python 3 version
+	// Bump this version when the version in omnibus/config/software/python3.rb changes
+	ExpectedPythonVersion3 = "3.11.5"
```
Comment (chouetz) [inline test/new-e2e/tests/agent-platform/common/agent_behaviour.go:201]: If we cannot read directly the value from the omnibus config, should we add a (pre) check to assess this (that expected versions are aligned)?

**Review by chouetz — APPROVED** [review]

File: test/new-e2e/tests/agent-platform/common/agent_behaviour.go (hunk 1 above)
Comment (pducolin) [reply test/new-e2e/tests/agent-platform/common/agent_behaviour.go:201]: Is there a job I could use ? Or better to add a new github action ?

Comment (pducolin) [reply test/new-e2e/tests/agent-platform/common/agent_behaviour.go:201]: Thinking about it, there are more spots where the python version is updated. If we want to automate that we don't miss anything, we should address this separately - see latest bump PR for an example https://github.com/DataDog/datadog-agent/pull/20218

Comment (pducolin) [reply test/new-e2e/tests/agent-platform/common/agent_behaviour.go:201]: Tracked by APL-2814

**Pull request merged**